openeogeotrellis/integrations/calrissian.py

```diff
@@ -408,7 +408,22 @@ def launch_job_and_wait(
         elif final_status is None:
             raise TimeoutError(f"CWL Job {job_name} did not finish within {timeout}s")
         elif final_status != "complete":
-            raise RuntimeError(f"CWL Job {job_name} failed with {final_status=} after {timer.elapsed()=:.2f}s")
+            # get logs from latest pod in job:
+            pods = kubernetes.client.CoreV1Api().list_namespaced_pod(
+                namespace=self._namespace, label_selector=f"job-name={job_name}", watch=False
+            )
+            last_pod = sorted(pods.items, key=lambda p: p.metadata.creation_timestamp, reverse=True)[0]
```

Comment:
don't `sort` to just get the last one, use `max`:
```suggestion
            last_pod = max(pods.items, key=lambda p: p.metadata.creation_timestamp)
```